h2o-algos/src/main/java/hex/modelselection/ModelSelection.java

```diff
@@ -227,7 +227,7 @@ void buildMaxRModels(ModelSelectionModel model) {
                         _parms, _foldColumn, _glmNFolds, _foldAssignment); // forward step
                 validSubset.removeAll(currSubsetIndices);
                 _job.update(predNum, "Finished building all models with "+predNum+" predictors.");
-                if (predNum < _numPredictors) {
+                if (predNum < _numPredictors && predNum > 1) {
                     GLMModel currBestR2Model = replacement(currSubsetIndices, coefNames, bestR2Model.r2(), _parms,
```

Comment:
When you have just one predictor, there is no need to switch it out and no need to call replacement step.